hack/minikube-rook-mirror-setup.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+set -x
+set -e -o pipefail
+
+## Variables
+PRIMARY_CLUSTER="${PRIMARY_CLUSTER:-hub}"
+SECONDARY_CLUSTER="${SECONDARY_CLUSTER:-cluster1}"
+POOL_NAME="replicapool"
+
+## Usage
+usage()
+{
+	set +x
+	echo "Usage:"
+	echo "  $0"
+	echo "  Available environment variables:"
+	echo "    minikube primary cluster PRIMARY_CLUSTER ${PRIMARY_CLUSTER}"
+	echo "    minikube secondary cluster SECONDARY_CLUSTER ${SECONDARY_CLUSTER}"
+	exit 0
+}
+
+function wait_for_condition() {
+    local count=15
+    local condition=${1}
+    local result
+    shift
+
+    while ((count > 0)); do
+        result=$("${@}")
+        if [[ "$result" == "$condition" ]]; then
+            return 0
+        fi
+        count=$((count - 1))
+        sleep 5
+    done
+
+    echo "Failed to meet $condition for command $*"
+    exit 1
+}
+
+SECONDARY_CLUSTER_PEER_TOKEN_SECRET_NAME=$(kubectl get cephblockpools.ceph.rook.io "${POOL_NAME}" --context="${SECONDARY_CLUSTER}" -nrook-ceph -o jsonpath='{.status.info.rbdMirrorBootstrapPeerSecretName}')
```

Comment:
I don't know why, but during one of my runs, this command didn't work causing the subsequent get secret command to fail.

$ ./minikube-rook-mirror-setup.sh
+ set -e -o pipefail
+ PRIMARY_CLUSTER=hub
+ SECONDARY_CLUSTER=cluster1
+ POOL_NAME=replicapool
++ kubectl get cephblockpools.ceph.rook.io replicapool --context=cluster1 -nrook-ceph -o 'jsonpath={.status.info.rbdMirrorBootstrapPeerSecretName}'
+ SECONDARY_CLUSTER_PEER_TOKEN_SECRET_NAME=
++ kubectl get secret -n rook-ceph '' --context=cluster1 -o 'jsonpath={.data.token}'
++ base64 -d
error: resource name may not be empty
+ SECONDARY_CLUSTER_SECRET=

Reply:
This was discussed as possibly missing the `--context=` in the script `minikube-rook-setup.sh` last line where the value of this field is checked, and now corrected.

Reply:
A run was performed with the updated `minikube-rook-setup.sh`, however the issue still occurred:
```
+ PROFILE=hub ramen/hack/minikube-rook-setup.sh create
++ kubectl get cephblockpool -n rook-ceph replicapool -o 'jsonpath={.status.info.rbdMirrorBootstrapPeerSecretName}' --context=hub
+ result=pool-peer-token-replicapool
+ [[ pool-peer-token-replicapool == \p\o\o\l\-\p\e\e\r\-\t\o\k\e\n\-\r\e\p\l\i\c\a\p\o\o\l ]]
+ return 0
+ echo Setup 'succesful!'

+ PROFILE=cluster1 ramen/hack/minikube-rook-setup.sh create
++ kubectl get cephblockpool -n rook-ceph replicapool -o 'jsonpath={.status.info.rbdMirrorBootstrapPeerSecretName}' --context=cluster1
+ result=pool-peer-token-replicapool
+ [[ pool-peer-token-replicapool == \p\o\o\l\-\p\e\e\r\-\t\o\k\e\n\-\r\e\p\l\i\c\a\p\o\o\l ]]
+ return 0
+ echo Setup 'succesful!'

+ PRIMARY_CLUSTER=hub SECONDARY_CLUSTER=cluster1 ramen/hack/minikube-rook-mirror-setup.sh
++ kubectl get cephblockpools.ceph.rook.io replicapool --context=cluster1 -nrook-ceph -o 'jsonpath={.status.info.rbdMirrorBootstrapPeerSecretName}'
+ SECONDARY_CLUSTER_PEER_TOKEN_SECRET_NAME=pool-peer-token-replicapool
++ kubectl get secret -n rook-ceph pool-peer-token-replicapool --context=cluster1 -o 'jsonpath={.data.token}'
++ base64 -d
+ SECONDARY_CLUSTER_SECRET=eyJmc2lkIjoiMjBkZmJjNWItODk1OS00YjdlLTk0ZTAtNWNlNjk3MjBjOWY2IiwiY2xpZW50X2lkIjoicmJkLW1pcnJvci1wZWVyIiwia2V5IjoiQVFDSlRKUmc4UEtTSUJBQURwNHRCejlOZlBaVzNyQm83dVJyNGc9PSIsIm1vbl9ob3N0IjoiW3YyOjE5Mi4xNjguMzkuNTc6MzMwMCx2MToxOTIuMTY4LjM5LjU3OjY3ODldIn0=

+ PRIMARY_CLUSTER=cluster1 SECONDARY_CLUSTER=hub ramen/hack/minikube-rook-mirror-setup.sh
++ kubectl get cephblockpools.ceph.rook.io replicapool --context=hub -nrook-ceph -o 'jsonpath={.status.info.rbdMirrorBootstrapPeerSecretName}'
+ SECONDARY_CLUSTER_PEER_TOKEN_SECRET_NAME=
++ kubectl get secret -n rook-ceph '' --context=hub -o 'jsonpath={.data.token}'
++ base64 -d
error: resource name may not be empty
+ SECONDARY_CLUSTER_SECRET=
```

The command was run shortly after and it produced the expected output:

```
$ kubectl get cephblockpools.ceph.rook.io replicapool --context=hub -nrook-ceph -o 'jsonpath={.status.info.rbdMirrorBootstrapPeerSecretName}'
pool-peer-token-replicapool
```

And for reference, the context was:

```
$ kubectl config current-context
hub
```

It seems the hub's secret name could be gotten, then could not, then could again.  One possible work-around is performing the get in a loop in `rook-mirror-setup` like the one in `rook-setup`.